File: uni_modules/tuikit-atomic-x/utssdk/app-android/kotlin/CallEngineManager.kt
package uts.sdk.modules.atomicx.kotlin

import android.content.Context
import com.tencent.cloud.tuikit.engine.call.TUICallEngine
import com.tencent.cloud.tuikit.engine.common.TUICommonDefine

object CallEngineManager {

    private const val TAG = "UTS-CallEngineManager: "

    /**
     * 初始化 TUICallEngine
     * @param context 上下文
     * @param sdkAppID 腾讯云 SDKAppID
     * @param userID 用户 ID
     * @param userSig 用户签名
     */
    fun initEngine(context: Context, sdkAppID: Int, userID: String, userSig: String) {
        TUICallEngine.createInstance(context).init(sdkAppID, userID, userSig, object : TUICommonDefine.Callback {
            override fun onSuccess() {
                Logger.i(TAG + "TUICallEngine init success")
            }

            override fun onError(code: Int, message: String?) {
                Logger.e(TAG + "TUICallEngine init failed, code: $code, message: ${message ?: ""}")
            }
        })
    }
}

File: uni_modules/tuikit-atomic-x/utssdk/app-ios/swift/CallEngineManager.swift
import DCloudUTSFoundation
import RTCRoomEngine

class CallEngineManager: NSObject {

    private static let TAG = "UTS-CallEngineManager: "

    public static let shared = CallEngineManager()

    /// 初始化 TUICallEngine
    /// - Parameters:
    ///   - sdkAppID: 腾讯云 SDKAppID
    ///   - userID: 用户 ID
    ///   - userSig: 用户签名
    public func initEngine(_ sdkAppID: Int32, userID: String, userSig: String) {
        TUICallEngine.createInstance().`init`(sdkAppID, userId: userID, userSig: userSig) {
            console.log(CallEngineManager.TAG, "TUICallEngine init success")
        } fail: { code, message in
            console.error(
                CallEngineManager.TAG,
                "TUICallEngine init failed, code: \(code), message: \(message ?? "")")
        }
    }
}
